backend/ssm/ssm.go

```diff
@@ -0,0 +1,76 @@
+package ssm
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/aws/aws-sdk-go/service/ssm/ssmiface"
+	"github.com/heetch/confita/backend"
+	"github.com/heetch/confita/ptr"
+	"strings"
+)
+
+type Backend struct {
+	client   ssmiface.SSMAPI
+	ssmPath string
+	cache    map[string][]byte
+}
+
+func NewBackend(ssm ssmiface.SSMAPI, path string) *Backend {
+	return &Backend{client: ssm, ssmPath: path}
+}
+
+func (b *Backend) Get(ctx context.Context, key string) ([]byte, error) {
+	if b.cache == nil {
+		err := b.fetchParams(ctx)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return b.fromCache(ctx, key)
+}
+
+func (b *Backend) Name() string {
+	return "ssm"
+}
+
+func (b *Backend) fetchParams(ctx context.Context) error {
+	b.cache = make(map[string][]byte)
+
+	ssmInput := &ssm.GetParametersByPathInput{
+		Path:           &b.ssmPath,
+		Recursive:      ptr.Bool(true),
+		WithDecryption: ptr.Bool(true),
+		MaxResults:     ptr.Int64(10),
```

Comment:
Perhaps use a larger value for this (e.g. 1000) to avoid unnecessary round trips when the number of configuration items is large?

Reply:
I wish we could. 1000 was my original number for your exact reason, however, 10 is the Amazon limit.  If you send any larger than that, you'll get a validation error from Amazon.